commands/azureCommands/delete-image.ts

```diff
@@ -14,34 +14,64 @@ import { Repository } from "../../utils/Azure/models/repository";
 
+/** Function to untag an Azure hosted image
+ * @param context : if called through right click on AzureImageNode, the node object will be passed in. See azureRegistryNodes.ts for more info
+ */
+export async function untagAzureImage(context?: AzureImageTagNode): Promise<void> {
+    await removeImage(context, true);
+}
+
 /** Function to delete an Azure hosted image
  * @param context : if called through right click on AzureImageNode, the node object will be passed in. See azureRegistryNodes.ts for more info
  */
 export async function deleteAzureImage(context?: AzureImageTagNode): Promise<void> {
+    await removeImage(context, false);
+}
+
+/** Function to delete an Azure hosted image
+ * @param context : if called through right click on AzureImageNode, the node object will be passed in. See azureRegistryNodes.ts for more info
+ * @param untag : if true deletes the image tag, otherwise removes digest and all other tags associeted to the image selected.
+ */
+async function removeImage(context: AzureImageTagNode, untag: boolean): Promise<void> {
+    let action: string = (untag) ? "untag" : "delete";
     let registry: Registry;
-    let repoName: string;
-    let tag: string;
+    let repo: Repository;
+    let image: AzureImage;
 
     if (!context) {
         registry = await quickPicks.quickPickACRRegistry();
-        const repository: Repository = await quickPicks.quickPickACRRepository(registry, 'Select the repository of the image you want to delete');
-        repoName = repository.name;
-        const image: AzureImage = await quickPicks.quickPickACRImage(repository, 'Select the image you want to delete');
-        tag = image.tag;
+        repo = await quickPicks.quickPickACRRepository(registry, `Select the repository of the image you want to ${action}`);
+        image = await quickPicks.quickPickACRImage(repo, `Select the image you want to ${action}`);
 
     } else {
         registry = context.registry;
         let wholeName: string[] = context.label.split(':');
-        repoName = wholeName[0];
-        tag = wholeName[1];
+        repo = await Repository.Create(registry, wholeName[0]);
+        image = new AzureImage(repo, wholeName[1]);
     }
 
-    const shouldDelete = await ext.ui.showWarningMessage(`Are you sure you want to delete ${repoName}:${tag}? `, { modal: true }, DialogResponses.deleteResponse, DialogResponses.cancel);
+    let message: string;
+    let path: string;
+    let digest: string;
+    if (untag) {
+        message = `Are you sure you want to untag: \'${image.toString()}\'? This does not delete the manifest referenced by the tag.`;
+        path = `/v2/_acr/${repo.name}/tags/${image.tag}`;
+    } else {
+        digest = await acrTools.getImageDigest(image);
+        let images = await acrTools.getImagesByDigest(repo, digest);
+        message = `Are you sure you want to delete the manifest: '${digest}' and the associated image(s) ${images.join(', ')}?`;
```

Comment:
![image](https://user-images.githubusercontent.com/6913354/48452776-72e31d00-e765-11e8-89e9-2c1f2ea66f13.png)
Nice, but do we really need the digest ID?  Is there a way we could word this better?

Reply:
I'm just copying the format on the Azure portal.
![image](https://user-images.githubusercontent.com/43052640/48453158-1254df80-e767-11e8-9ee1-a24b3c5989bc.png)